pkg/upgrade/upgrade.go

```diff
@@ -147,13 +145,27 @@ func Upgrade(flagVersion string, log log.Logger) error {
 		return nil
 	}
 
-	newerVersion, err := CheckForNewerVersion()
+	latestStable, err := CheckForNewerVersion()
 	if err != nil {
 		return err
 	}
-	if newerVersion == "" {
+	if latestStable == "" {
+		log.Infof("Current binary is the latest version: %s", version)
+		return nil
+	}
+
+	version = GetVersion()
+	semverCurrentVersion := semver.MustParse(version)
+	semverLatestStableVersion := semver.MustParse(latestStable)
```

Comment:
Can this not include a `v` and fail?

Reply:
No, neither of these are from user input.  `latestStable` orginates from a `semver.Version.String()`